### PR TITLE
[kube-prometheus-stack] Fix alertmanager podAntiAffinity

### PR DESCRIPTION
**Pull request by CVirus**
#### What this PR does / why we need it:

This PR fixes the `matchExpression` for alertmanager `podAntiAffinity` configuration. Replacing `app` with `app.kubernetes.io/name` since alertmanager pods do NOT have an `app` label so the podAntiAffinity config didn't actually work.

#### Checklist
- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by CVirus** [comment]
@andrewgkew @bismarck @desaintmartin @gianrubio @gkarthiks @scottrigby @Xtigyro Please review whenever you find. Thanks in advance!

**Review by monotek — APPROVED** [review]

**Comment by CVirus** [comment]
@monotek Thanks for the approval. Do I need approval from ALL the maintainers to get this merged?

**Review by mrueg — APPROVED** [review]
not a maintainer but this makes sense to me.

File: charts/kube-prometheus-stack/Chart.yaml
Comment (monotek) [inline charts/kube-prometheus-stack/Chart.yaml:24]:
```suggestion
version: 30.0.3
```

**Review by monotek — CHANGES_REQUESTED** [review]

**Review by monotek — APPROVED** [review]

**Comment by CVirus** [comment]
@monotek @mrueg Sorry had to sync fork. Please approve again whenever you can.

**Review by mrueg — APPROVED** [review]

**Pull request merged**
